pulp_container/app/models.py

```diff
@@ -382,6 +405,45 @@ def get_downloader(self, remote_artifact=None, url=None, **kwargs):
         kwargs["remote"] = self
         return super().get_downloader(remote_artifact=remote_artifact, url=url, **kwargs)
 
+    def get_noauth_downloader(self, remote_artifact=None, url=None, **kwargs):
```

Comment:
The only difference from the pulpcore method is the download factory which is used. I suggest to change pulpcore method to accept an optional parameter `download_factory` and keep self.download_factory` as default.